scm-step/src/main/java/org/jenkinsci/plugins/workflow/steps/scm/GitStep.java

```diff
@@ -43,6 +65,7 @@
 
     private final String url;
     private String branch = "master";
+    private StandardCredentials credentials;
```

Comment:
No, you cannot store a `Credentials` like this, you should only keep the `credentialsId` as you had it before (not sure why you changed it).